### PR TITLE
Fixed a11y violations for Slider examples

### PR DESCRIPTION
**Pull request by rohan-kadkol**
## Changes

Added `aria-label` prop to all thumbs in all `Slider` examples. But no change was needed in component code.

## Testing

Ensured that the cypress tests for Slider reported no more violations.

## Docs

N/A

### CODE REVIEW

**Review by mayank99 — COMMENTED** [review]
can we try to use a visible label in most examples? will likely need `aria-labelledby` instead of `aria-label`

**Comment by rohan-kadkol** [comment]
> can we try to use a visible label in most examples? will likely need `aria-labelledby` instead of `aria-label`

I adding `<Text as="label">...</Text>` to most examples for `<Slider>`. I linked the slider to the labels using `aria-labeledby` for these cases, and `aria-label` when not using the explicit labels. (https://github.com/iTwin/iTwinUI/pull/1474/commits/0b444e0afb722016df7d45f157def5951ffd93e7)

File: examples/Slider.labels.tsx [hunk 1]
```diff
@@ -3,19 +3,25 @@
  * See LICENSE.md in the project root for license terms and full copyright notice.
  *--------------------------------------------------------------------------------------------*/
 import * as React from 'react';
-import { Slider } from '@itwin/itwinui-react';
+import { Slider, Text } from '@itwin/itwinui-react';
+import { useId } from '../packages/itwinui-react/src/core/utils/hooks/index.js';
```
Comment (mayank99) [inline examples/Slider.labels.tsx]: use `React.useId` instead of this internal hook

File: examples/Slider.labels.tsx [hunk 2]
```diff
@@ -3,19 +3,25 @@
...
   SvgSmileyHappyVery,
   SvgSmileySadVery,
 } from '@itwin/itwinui-icons-react';
 
 export default () => {
+  const labelId = useId();
+
   return (
     <div style={{ width: 'min(100%, 300px)' }}>
+      <Text id={labelId} as='label' variant='leading'>
```
Comment (mayank99) [inline examples/Slider.labels.tsx:18]: `<Label>` styling should work better than `<Text variant='leading'>`.

also not sure if we should use `as='label'` or `as='div'`.

File: examples/Slider.labels.tsx [hunk 3]
```diff
@@ -3,19 +3,25 @@
...
+        Choose a happiness level
+      </Text>
       <Slider
         thumbProps={() => {
           return {
-            'aria-label': `Choose a rating`,
+            'aria-labelledby': labelId,
```
Comment (mayank99) [inline examples/Slider.labels.tsx:24]: nit: the `return` can be condensed using parentheses

```
thumbProps={() => ({
  'aria-labelledby': labelId
})}
```

Comment (rohan-kadkol) [reply examples/Slider.labels.tsx]: Ohh, right, changed for all Slider examples with 1-2 keys in the `thumbProps` object (https://github.com/iTwin/iTwinUI/pull/1474/commits/65f28ded5083045369be3ac44799fc272593722f)

File: examples/Slider.tooltipnone.tsx [hunk 4]
```diff
@@ -26,7 +28,14 @@ export default () => {
 
   return (
     <div style={{ width: 'min(100%, 300px)' }}>
+      <Text id={labelId} as='label' variant='leading'>
+        Choose a start date
+      </Text>
       <Slider
+        thumbProps={() => ({
+          'aria-labelledby': labelId,
+          'aria-valuetext': dateFormatter.format(currentValue.date),
```
Comment (mayank99) [inline examples/Slider.tooltipnone.tsx:37]: i feel like `aria-valuetext` should not need to be set by users. why was this needed?

Comment (rohan-kadkol) [reply examples/Slider.tooltipnone.tsx:37]: This was not needed (did not cause any a11y violation). But since the screen reader would just read out the date number (e.g. `35`), I thought of adding `aria-valuetext` with the formatted date string (e.g. something like `4 Feb`)

File: examples/Slider.labels.tsx (hunk 2 above)
Comment (rohan-kadkol) [reply examples/Slider.labels.tsx]: > `<Label>` styling should work better than `<Text variant='leading'>`.

https://github.com/iTwin/iTwinUI/pull/1474/commits/0eb23657b5e8e0d71dc948c55cc879621677db8e: Ohh, nice. Yes, `<Label>` styling looks great. It also now looks more similar to the label in LabeledSelect.

> also not sure if we should use `as='label'` or `as='div'`.

https://github.com/iTwin/iTwinUI/pull/1474/commits/03ac73f11df68437e7a7cdf8f20b1ee6248596e8: Since the slider is actually a `<div role="slider">` and not an `<input type="range">`, I changed the `as` of `<Label>` to be a `div` and not a `label`. I found [this example on the MDN docs](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/slider_role#examples) to confirm this pattern of:

| Label   | Slider|
|----------|-------------|
| `<label for="sliderId">` |  `<input id="sliderId" type="range">`  |
| **`<div id="labelId">`** | **`<div role="slider">`** (our iTwinUI Slider) |

**Review by mayank99 — APPROVED** [review]
